MechanicsCore/src/main/java/me/deecaad/core/mechanics/conditions/MythicMobsFactionCondition.java

```diff
@@ -0,0 +1,59 @@
+package me.deecaad.core.mechanics.conditions;
+
+import io.lumine.mythic.api.adapters.AbstractEntity;
+import io.lumine.mythic.bukkit.BukkitAdapter;
+import io.lumine.mythic.bukkit.MythicBukkit;
+import io.lumine.mythic.core.mobs.ActiveMob;
+import me.deecaad.core.file.SerializeData;
+import me.deecaad.core.file.SerializerException;
+import me.deecaad.core.mechanics.CastData;
+import me.deecaad.core.mechanics.conditions.Condition;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Optional;
+
+public class MythicMobsFactionCondition extends Condition {
+
+    private String faction;
+
+    public MythicMobsFactionCondition() {}
+
+    public MythicMobsFactionCondition(String faction) {
+        this.faction = faction;
+    }
+
+    @Override
+    protected boolean isAllowed0(CastData cast) {
+        LivingEntity target = cast.getTarget();
+        if (target == null) return false;
+
+        if (target instanceof Player) {
```

Comment:
You should be using the [enhanced instanceof](https://www.baeldung.com/java-pattern-matching-instanceof#pattern-matching) from Java 14